text/0000-angle-bracket-invocation.md

```diff
@@ -0,0 +1,259 @@
+be normalized using the `dasherize` function, which is the same rules used by
+existing use cases, such as service injections. This allows existing components
+to be invoked by the new syntax.
+
+Another benefit of the captial letter rule is that we can now support component
+names with a single word, such as `<Button>`, `<Modal>` and `<Tab>`.
+
+> Note: Some day, we may want to explore a file system migration to remove the
+> need for the normalization rule (i.e. also use capital case in filenames).
+> However, that is out-of-scope for this RFC, as it would require taking into
+> consideration existing code (like services), transition paths and codemods.
+
+### Arguments
+
+The next part of the invocation is passing arguments to the invoked component.
+We propose to use the `@` syntax for this purpose. For example, the invocation
+`<FooBar @foo=... @bar=... />` is equivilant to `{{foo-bar foo=... bar=...}}`.
+This matches the [named arguments syntax](https://github.com/emberjs/rfcs/pull/276)
+in the component template.
+
+If the argument value is a constant string, it can appear verbatim after the
+equal sign, i.e. `<FooBar @foo="some constant string" />`. Other values should
+be enclosed in curlies, i.e. `<FooBar @foo={{123}} @bar={{this.bar}} />`.
+Helpers can also be used, as in `<FooBar @foo={{capitalize this.bar}} />`.
+
+#### Reserved Names
+
+`@args`, `@arguments` and anything that does not start with a lowercase letter
+(such as `@Foo`, `@0`, `@!` etc) are reserved names and cannot be used. These
+restrictions may be relaxed in the future.
+
+#### Positional Arguments
+
+Positional arguments (`{{foo-bar "first" "second"}}`) are not supported.
+
+### HTML Attributes
+
+HTML attributes can be passed to the component using the regular HTML syntax.
+For example, `<FooBar class="btn btn-large" role="button" />`. HTML attributes
+can be interleaved with named arguments (it does not make any difference). This
+is a new feature that is not available in the classic invocation style.
+
+These attributes can be accessed from the component template with the new
+`...attributes` syntax, which is available only in element positions, e.g.
+`<div ...atributes />`. Using `...attributes` in any other positions, e.g.
+`<div>{{...attributes}}</div>`, would be a syntax error. It can also be used on
+multiple elements in the same template. If attributes are passed but the
+component template does not contain `...attributes` (i.e. the invoker passed
+some attributes, but the invokee does not take them), it will be a development
+mode error.
+
+It could be thought of that the attributes in the invocation side is stored in
+an internal block, and `...attributes` is the syntax for yielding to this
+internal block. Since the `yield` keyword is not available in element position,
+a dedicated syntax is needed.
+
+> Open Question: Should classic components (`Ember.Component`) implicitly add
```

Comment:
I'm assuming 'classic component' is a reference to the component manager RFC? Initially this paragraph was (maybe still is) very confusing to me. I guess I can understand the point of the question if I assume the following:

1. 'Classic components' (current-day Ember components) do not support `...attributes`, so throwing an error when they are not used seems wrong (at least initially)
2. Glimmer components do support it so they will work as described in the paragraphs above

Can someone please correct my understanding if it's wrong?

Reply:
Yes, that seems correct.

I think that classic components should fully support `...attributes` and that we should automatically insert `...attributes` to the wrapping element if it doesn't appear in the template and `tagName: ''` hasn't been used.

This would allow people to immediately start using HTML attributes with their classic components and remove the need for hacky add-ons like https://github.com/mmun/ember-component-attributes.